### PR TITLE
Add playbook for gzipping logs

### PR DESCRIPTION
**Pull request by MSevey**
# PULL REQUEST

## Overview
Since logrotate never worked and we keep seeing disk space issues I made a helper ansible script that we can use to gzip various logs. The log files are defined with their absolute paths in the `portal_log_files` variable, so we can easily add logs. The current size trigger is 5GB which can easily be updated as well.

## Example for Visual Changes

Initial test on dev1 with 1GB as the limit
Screen Shot 2021-11-24 at 1 14 42 PM

Verified on dev3
Screen Shot 2021-11-24 at 1 42 45 PM

Screen Shot 2021-11-24 at 1 44 06 PM

## Checklist
Review and complete the checklist to ensure that the PR is complete before assigned to an approver.
 - [x] All new methods or updated methods have clear docstrings
 - [x] Testing added or updated for new methods
 - [x] Verify if any changes impact the WebPortal Health Checks
 - [x] Approriate documentation updated
 - [x] Changelog file created

### CODE REVIEW

**Comment by DavidVorick** [comment]
I don't think we need to keep most of these logs. Can you keep the nginx` access.log` files out of the script? I'll set up a separate thing to compress them as part of my metrics manager.

File: playbooks/tasks/portal-gzip-logs.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+# Stat all the portal_log_files.
+- name: Stat Files
+  ansible.builtin.stat:
+    path: "{{ file }}"
+  register: loop_results
+  loop: "{{ portal_log_files }}"
+  loop_control:
+    loop_var: file
+  when: portal_log_files is defined
```
Comment (firyx) [inline playbooks/tasks/portal-gzip-logs.yml:11]: I would leave out this `when`. If `portal_log_files` are not defined user gets a message of the missing config which should be mandatory for this playbook.

File: playbooks/tasks/portal-gzip-logs.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  become: True
+
+# GZIP any files over the portal_log_size_gzip_limit
+- name: GZIP Files
+  community.general.archive:
+    path: "{{ file.stat.path }}"
+    dest: "{{ file.stat.path + lookup('pipe','date +%Y-%m-%dT%H:%M:%S') + '.gz' }}"
+    remove: yes
+  loop: "{{ loop_results.results | from_yaml | list }}"
```
Comment (firyx) [inline playbooks/tasks/portal-gzip-logs.yml:20]: nit: I think `| from_yaml | list` is not needed, `loop_results.results` should be already a list.

**Review by firyx — CHANGES_REQUESTED** [review]

File: playbooks/portals-gzip-logs.yml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      include_tasks: tasks/portal-health-checks-run.yml
+
+    # Takedown portals in takedown group
+    - block:
+        # Include getting wanted docker-compose files
+        - name: Include getting wanted docker-compose files
+          include_tasks: tasks/portal-docker-compose-files-get-wanted.yml
+
+        # Start just sia docker service
+        - name: Start just sia docker service
```
Comment (firyx) [inline playbooks/portals-gzip-logs.yml:56]: Sia is already running, should this be moved up?
Also skyd now requires mongo.

**Review by firyx — CHANGES_REQUESTED** [review]

File: playbooks/tasks/portal-gzip-logs.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  register: loop_results
+  loop: "{{ portal_log_files }}"
+  loop_control:
+    loop_var: file
+  when: portal_log_files is defined
+  become: True
+
+# GZIP any files over the portal_log_size_gzip_limit
+- name: GZIP Files
+  community.general.archive:
```
Comment (firyx) [inline playbooks/tasks/portal-gzip-logs.yml:16]: Seems to need `become: True` to modify existing log files.

**Review by firyx — CHANGES_REQUESTED** [review]

File: playbooks/tasks/portal-gzip-logs.yml (hunk 2 above)
Comment (MSevey) [reply playbooks/tasks/portal-gzip-logs.yml:20]: it is a list, but not a nicely formatted one. This was from a StackOverflow post.  If you try and loop over it and print it out it is all funky,

File: playbooks/group_vars/webportals.yml [hunk 5]
```diff
@@ -177,3 +177,16 @@ docker_services_exclude_waiting:
 local_logs_dir: "../my-logs"
 # Path relative to ./playbooks/vars directory
 private_vars_file: "../../../ansible-private/private-vars/private-vars.yml"
+
+# Portal log file paths
+portal_log_files:
```
Comment (firyx) [inline playbooks/group_vars/webportals.yml:182]: Should use:
`- "{{ webportal_dir }}/docker/data/..."`

**Review by firyx — CHANGES_REQUESTED** [review]

**Review by firyx — CHANGES_REQUESTED** [review]
We should make gzipping logs optional part of portal deploy or portal restart, same as `clean_renter_log` is used here:
https://github.com/SkynetLabs/ansible-playbooks/blob/master/playbooks/tasks/portal-stop.yml#L14

Gzipping should be done via Alpine docker container so we don't need to add sudo (LastPass login) to portal restart/deploy playboks like here:
https://github.com/SkynetLabs/ansible-playbooks/blob/master/playbooks/tasks/portal-logs-clean-renter-log.yml

**Pull request closed without merging**